kt/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/filebuilder/MainEntryFileBuilder.kt

```diff
@@ -48,4 +60,21 @@ object MainEntryFileBuilder {
         initFunctionSpec.addStatement(templateString, *templateArgs)
         return this
     }
+
+    fun registerUserTypesVariantMappings(registeredClasses: List<RegisteredClass>) {
+        registerUserTypesVariantMappingsFunSpec.addStatement(
+            "val ret = %M<%T<*>>()",
+            MemberName("kotlin.collections", "mutableListOf"),
+            ClassName("kotlin.reflect", "KClass")
+        )
+        for (registeredClass in registeredClasses) {
+            registerUserTypesVariantMappingsFunSpec.addStatement(
+                "ret.add(%T::class)",
+                ClassName(registeredClass.containingPackage, registeredClass.name)
+            )
+        }
+        registerUserTypesVariantMappingsFunSpec.addStatement(
+            "return ret"
+        )
+    }
```

Comment:
This could all be a single return statement of an immutable list if you pass in the classes in the constructor of the list.
So the generated code would look like:
```kotlin
return listOf(
    ClassOne::class,
    ClassTwo::class
)
```

So probably the generation code would look something like this (probably not completely correct as I'm just typing that in the browser):
```kotlin
val codeBlock = CodeBlock
    .builder()
    .addStatement("return %M(", MemberName("kotlin.collections", "listOf"))

registeredClasses.forEach { registeredClass ->
    codeBlock.addStatement("%T::class,", ClassName(registeredClass.containingPackage, registeredClass.name))
}
registerUserTypesVariantMappingsFunSpec.addCode(codeBlock.build())
```